File: src/main/java/com/distribution/modules/job/service/impl/ScheduleJobLogServiceImpl.java
package com.distribution.modules.job.service.impl;

import com.distribution.modules.job.dao.ScheduleJobLogDao;
import com.distribution.modules.job.entity.ScheduleJobLogEntity;
import com.distribution.modules.job.service.ScheduleJobLogService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;
import java.util.Map;

@Service("scheduleJobLogService")
public class ScheduleJobLogServiceImpl implements ScheduleJobLogService {
    @Autowired
    private ScheduleJobLogDao scheduleJobLogDao;

    @Override
    public ScheduleJobLogEntity queryObject(Long jobId) {
        return scheduleJobLogDao.queryObject(jobId);
    }

    @Override
    public List<ScheduleJobLogEntity> queryList(Map<String, Object> map) {
        return scheduleJobLogDao.queryList(map);
    }

    @Override
    public int queryTotal(Map<String, Object> map) {
        return scheduleJobLogDao.queryTotal(map);
    }

    @Override
    @Transactional(rollbackFor = Exception.class)
    public void save(ScheduleJobLogEntity log) throws Exception {
        scheduleJobLogDao.save(log);
    }

}

File: src/main/java/com/distribution/modules/job/service/impl/ScheduleJobLogServiceImpl.kt
package com.distribution.modules.job.service.impl

import com.distribution.modules.job.dao.ScheduleJobLogDao
import com.distribution.modules.job.entity.ScheduleJobLogEntity
import com.distribution.modules.job.service.ScheduleJobLogService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

@Service("scheduleJobLogService")
class ScheduleJobLogServiceImpl : ScheduleJobLogService {
    @Autowired
    private lateinit var scheduleJobLogDao: ScheduleJobLogDao

    override fun queryObject(jobId: Long): ScheduleJobLogEntity {
        return scheduleJobLogDao.queryObject(jobId)
    }

    override fun queryList(map: Map<String, Any>): List<ScheduleJobLogEntity> {
        return scheduleJobLogDao.queryList(map)
    }

    override fun queryTotal(map: Map<String, Any>): Int {
        return scheduleJobLogDao.queryTotal(map)
    }

    @Transactional(rollbackFor = [(Exception::class)])
    @Throws(Exception::class)
    override fun save(log: ScheduleJobLogEntity) {
        scheduleJobLogDao.save(log)
    }

}
